components/main_menu/main_menu.tsx

```diff
@@ -147,6 +160,7 @@ export class MainMenu extends React.PureComponent<Props> {
 
         const someIntegrationEnabled = this.props.enableIncomingWebhooks || this.props.enableOutgoingWebhooks || this.props.enableCommands || this.props.enableOAuthServiceProvider || this.props.canManageSystemBots;
         const showIntegrations = !this.props.mobile && someIntegrationEnabled && this.props.canManageIntegrations;
+        const createTeamRestricted = this.props.isCloudFreeEnabled && (this.props.isFreeTrial || this.props.teamsLimitReached);
```

Comment:
Here we need to check if they are on a cloud license, the feature flag will be removed once we launch and this checks should not apply to self-managed servers